Std/Data/List/Init/Lemmas.lean

```diff
@@ -15,15 +19,12 @@ These are theorems used in the definitions of `Std.Data.List.Basic`.
 New theorems should be added to `Std.Data.List.Lemmas` if they are not needed by the bootstrap.
 -/
 
-attribute [simp] get reverseAux eraseIdx map join find? findSome?
-  replace elem lookup drop take foldl foldr zipWith unzip range.loop enumFrom
-  intersperse isPrefixOf isEqv dropLast iota mapM.loop mapA List.forM forA filterAuxM
-  filterMapM.loop List.foldlM firstM anyM allM findM? findSomeM? forIn.loop forIn'.loop
-  concat_eq_append append_assoc
```

Comment:
These have not been replaced with the appropriate simp lemmas. For example, instead of `attribute [simp] reverseAux`, there should be:
```lean
@[simp] theorem reverseAux_nil : reverseAux [] r = r := rfl
@[simp] theorem reverseAux_cons : reverseAux (a::l) r = reverseAux l (a::r) := rfl
```
These should also be in `Data.List.Init.Lemmas`, not `Data.List.Lemmas` because the equation lemmas are considered part of the definition itself.